eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/DisallowedWordsCheck.java

```diff
@@ -0,0 +1,152 @@
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.CLASS_DEF,
+        TokenTypes.VARIABLE_DEF,
+        TokenTypes.METHOD_DEF};
+    }
+
+    @Override
+    public void visitToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.CLASS_DEF:
+            case TokenTypes.METHOD_DEF:
+            case TokenTypes.VARIABLE_DEF:
+                String tokenName = token.findFirstToken(TokenTypes.IDENT).getText();
+                if (shouldCheckInScope(token)) {
+                    String result = getDisallowedWords(tokenName);
+                    if (result != null) {
+                        log(token, String.format(errorMessage, tokenName, this.disallowedWords.stream().collect(Collectors.joining(", ", "", ""))));
+                    }
+                }
+                break;
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     * Should we check member with given modifiers.
+     *
+     * @param token
+     *                modifiers of member to check.
+     * @return true if we should check such member.
+     */
+    private boolean shouldCheckInScope(DetailAST token) {
+        final DetailAST modifiersAST =
+            token.findFirstToken(TokenTypes.MODIFIERS);
+        final boolean isPublic = modifiersAST
+            .findFirstToken(TokenTypes.LITERAL_PUBLIC) != null;
+        return applyToPublic && isPublic;
+    }
+
+    /**
+     * Gets the disallowed abbreviation contained in given String.
+     * @param tokenName
+     *        the given String.
+     * @return the disallowed abbreviation contained in given String as a
+     *         separate String.
+     */
+    private String getDisallowedWords(String tokenName) {
+        int beginIndex = 0;
+        boolean abbrStarted = false;
+        String result = null;
+
+        for (int index = 0; index < tokenName.length(); index++) {
+            final char symbol = tokenName.charAt(index);
+
+            if (Character.isUpperCase(symbol)) {
+                if (!abbrStarted) {
+                    abbrStarted = true;
+                    beginIndex = index;
+                }
+            }
+            else if (abbrStarted) {
+                abbrStarted = false;
+
+                final int endIndex = index - 1;
+                result = getAbbreviationIfIllegal(tokenName, beginIndex, endIndex);
+                if (result != null) {
+                    break;
+                }
+                beginIndex = -1;
+            }
+        }
+        // if abbreviation at the end of name (example: scaleX)
+        if (abbrStarted) {
+            final int endIndex = tokenName.length() - 1;
+            result = getAbbreviationIfIllegal(tokenName, beginIndex, endIndex);
+        }
+        return result;
```

Comment:
This whole method could potentially be simplified to use the Java String `contains` method looping over the set of disallowed words?

Reply:
Good call. The code I used is from a checkstyle class but this is definitely a better solution.